arangod/Graph/BaseOptions.cpp

```diff
@@ -658,6 +688,39 @@ void BaseOptions::parseShardIndependentFlags(arangodb::velocypack::Slice info) {
   setMaxProjections(VPackHelper::getNumericValue<size_t>(
       info, StaticStrings::MaxProjections,
       DocumentProducingNode::kMaxProjections));
-  _vertexProjections = Projections::fromVelocyPack(info, "vertexProjections");
-  _edgeProjections = Projections::fromVelocyPack(info, "edgeProjections");
-}
+
+  {
+    if (!getVertexProjections().empty()) {
+      resourceMonitor().decreaseMemoryUsage(
+          getVertexProjections().size() * sizeof(aql::Projections::Projection));
+    }
+
+    _vertexProjections = Projections::fromVelocyPack(info, "vertexProjections");
+    try {
+      arangodb::ResourceUsageScope vGuard(
+          resourceMonitor(),
+          getVertexProjections().size() * sizeof(aql::Projections::Projection));
+      vGuard.steal();  // now we are responsible for tracking the memory
+    } catch (...) {
+      _vertexProjections.clear();
+      throw;
+    }
+  }
+
+  {
+    if (!getEdgeProjections().empty()) {
+      resourceMonitor().decreaseMemoryUsage(
+          getEdgeProjections().size() * sizeof(aql::Projections::Projection));
+    }
+    _edgeProjections = Projections::fromVelocyPack(info, "edgeProjections");
+    try {
+      arangodb::ResourceUsageScope eGuard(
+          resourceMonitor(),
+          getEdgeProjections().size() * sizeof(aql::Projections::Projection));
+      eGuard.steal();  // now we are responsible for tracking the memory
+    } catch (...) {
+      _edgeProjections.clear();
```

Comment:
same as above

Reply:
Added in: 60044d169796e5dd8850ba2454d5f44463b1fc76